app/grandchallenge/algorithms/models.py

```diff
@@ -103,6 +103,13 @@ class Algorithm(UUIDModel, TitleSlugDescriptionModel, ViewContentMixin):
         blank=True,
         on_delete=models.SET_NULL,
     )
+    optional_hanging_protocols = models.ManyToManyField(
+        "hanging_protocols.HangingProtocol",
+        through="OptionalProtocolsAlgorithm",
+        related_name="optional_of_algorithm",
```

Comment:
I don't think that "of" is the correct preposition, IMO it should be "for".

`HangingProtocol.objects.filter(optional_for_algorithm=algorithm)` => Give me the hanging protocols that are optional for this algorithm.

Same for the other apps.